### ISSUE TITLE
doctest failures on Prefix [9.5.beta8]

### ISSUE THREAD

**Issue by strogdon** [cschwan/sage-on-gentoo#664]
I have these failures
```
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py  # 1 doctest failed
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/rings/complex_arb.pyx  # 3 doctests failed
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/functions/gamma.py  # 1 doctest failed
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/symbolic/function.pyx  # 1 doctest failed
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/rings/real_arb.pyx  # 12 doctests failed
```
The last 4 appear to be associated with`numerical noise`. However, the first has failure
```
sage -t --long --warn-long 168.3 --random-seed=149786964115744708076534245704671867774 usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py
**********************************************************************
File "usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 1138, in sage.schemes.cyclic_covers.cycliccover_finite_field.CyclicCover_finite_field.frobenius_polynomial
Failed example:
    CyclicCover(11, PolynomialRing(GF(1129), 'x')([-1] + [0]*(5-1) + [1])).frobenius_polynomial() # long time
Exception raised:
    Traceback (most recent call last):
      File "sage/misc/cachefunc.pyx", line 1943, in sage.misc.cachefunc.CachedMethodCaller.__call__ (/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_9/cythonized/sage/misc/cachefunc.c:10383)
        return cache[k]
    KeyError: ((11,), ())

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/doctest/forker.py", line 694, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/doctest/forker.py", line 1088, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.schemes.cyclic_covers.cycliccover_finite_field.CyclicCover_finite_field.frobenius_polynomial[23]>", line 1, in <module>
        CyclicCover(Integer(11), PolynomialRing(GF(Integer(1129)), 'x')([-Integer(1)] + [Integer(0)]*(Integer(5)-Integer(1)) + [Integer(1)])).frobenius_polynomial() # long time
      File "sage/misc/cachefunc.pyx", line 2310, in sage.misc.cachefunc.CachedMethodCallerNoArgs.__call__ (/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_9/cythonized/sage/misc/cachefunc.c:12993)
        self.cache = f(self._instance)
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 1233, in frobenius_polynomial
        F = self.frobenius_matrix(self._N0)
      File "sage/misc/cachefunc.pyx", line 1948, in sage.misc.cachefunc.CachedMethodCaller.__call__ (/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_9/cythonized/sage/misc/cachefunc.c:10519)
        w = self._instance_call(*args, **kwds)
      File "sage/misc/cachefunc.pyx", line 1824, in sage.misc.cachefunc.CachedMethodCaller._instance_call (/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_9/cythonized/sage/misc/cachefunc.c:9985)
        return self.f(self._instance, *args, **kwds)
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 1071, in frobenius_matrix
        FrobP = _frobenius_matrix_p(self._N0)
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 1067, in _frobenius_matrix_p
        ] = self._frob(i, j + self._epsilon * self._r, N0)
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 993, in _frob
        self._initialize_fat_horizontal(
      File "/storage/strogdon/gentoo-rap/usr/lib/python3.9/site-packages/sage/schemes/cyclic_covers/cycliccover_finite_field.py", line 737, in _initialize_fat_horizontal
        DH = [elt[0, 0] for elt in interval_products(D0, D1, targets)]
      File "sage/schemes/hyperelliptic_curves/hypellfrob.pyx", line 140, in sage.schemes.hyperelliptic_curves.hypellfrob.interval_products (/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_9/cythonized/sage/schemes/hyperelliptic_curves/hypellfrob.cpp:3983)
        sig_on()
    SystemError: calling remove_from_pari_stack() inside sig_on()
```

Which may be the real source of the other failures? I suspect an `unstable` component has been inadvertently installed.

**Comment by strogdon** [comment]
The `cycliccover_finite_field.py` did pass when run individually.

**Comment by kiwifb** [comment]
`arb` would be the most obvious suspect along with `flint`.

**Comment by strogdon** [comment]
`flint` is OK. Downgrading `arb` to `2.19.0` from `2.21.0` fixed the latter 4 failures. Not sure about the first one involving `cycliccover_finite_field.py` which passes individually. I close.

**Comment by kiwifb** [comment]
`arb` version is now pinned in 9.4 and 9999.

**Comment by slel** [comment]
Sadly https://trac.sagemath.org/ticket/32905
which adds a tolerance indication for these doctests
was not merged in Sage 9.5.beta8.
